### PR TITLE
Unit tests for serving trays

### PR DESCRIPTION
**Pull request by nightred**
## About The Pull Request

Created some tests to validate that our serving tray boy 
* can pick up some foods
* put down what it carries on a table
* can be put in pockets
* drops items when smacking people

## Why It's Good For The Game

More unit tests

## Changelog
Ya skipping this it is a unit test

### CODE REVIEW

File: code/modules/unit_tests/servingtray.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	var/obj/item/storage/bag/tray/test_tray = allocate(/obj/item/storage/bag/tray)
+	var/obj/item/reagent_containers/food/banana = allocate(/obj/item/reagent_containers/food/snacks/grown/banana)
+	var/obj/item/food/the_bread = allocate(/obj/item/food/breadslice)
+	var/obj/item/reagent_containers/food/sugarcookie = allocate(/obj/item/reagent_containers/food/snacks/sugarcookie)
+
+	TEST_ASSERT_EQUAL((the_bread in test_tray.contents), FALSE, "The bread is on the serving tray at test start")
+
+	// set the tray to single item mode the dirty way
+	var/datum/component/storage/tray_storage = test_tray.GetComponent(/datum/component/storage)
+	tray_storage.collection_mode = 0 //COLLECT_ONE
```
Comment (Jared-Fogle) [inline code/modules/unit_tests/servingtray.dm:16]: Can you move out `COLLECT_ONE` and the other defines to something like `code/__DEFINES/storage.dm`? I'm uncomfortable relying on the value here.

Comment (nightred) [reply code/modules/unit_tests/servingtray.dm:16]: Fair, i will add this tot he next update where it passes it's own test

**Review by Mothblocks — DISMISSED** [review]

**Comment by Jared-Fogle** [comment]
One last nitpick--can you rename it to serving_tray.dm for consistency's sake?

**Comment by nightred** [comment]
The great renaming has been done.

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**